### PR TITLE
Update relevant documentation and manifests to reflect the changes made to authProxy.

### PR DESCRIPTION
**Pull request by timflannagan1**
This would update documentation surrounding viewing reports and configuring the reporting operator. PR #814 would enable the metering route by default, which allows us to remove the `manifests/metering-config/expose-route.yaml` file as it's no longer needed.

The wording and the organization of the sections most likely need to be improved.

I was considering adding the `reporting-operator.spec.authProxy.htpasswd` section to `manifests/metering-config/advanced-auth.yaml` as I think it still has a place there.

I also think that it might be helpful to describe which keys/fields of the `reporting-operator.spec.authProxy` are enabled by default.

### CODE REVIEW

File: Documentation/configuring-reporting-operator.md [hunk 1]
```diff
@@ -147,6 +119,46 @@ curl "http://35.227.172.86:8080/api/v1/reports/get?name=cluster-memory-capacity-
...
 ### Openshift Authentication
 
+By default, the reporting API is secured with TLS and authentication. This is done by configuring the reporting-operator to deploy a pod containing both the reporting-operator's container, and a sidecar container running [Openshift auth-proxy](https://github.com/openshift/oauth-proxy).
+
+If you want to manually configure authentication in Openshift, disable it entirely, or want more information on the reporting-operator options, see the [manual setup](#manual-setup) section.
+
+In order to access the reporting API, the metering operator exposes a route. Once that route has been installed, you can run the command below to get the route's hostname.
+
+```
+METERING_ROUTE_NAME=$(oc -n $METERING_NAMESPACE get routes metering -o json | jq '.status.ingress[].host')
```
Comment (chancez) [inline Documentation/configuring-reporting-operator.md]: Maybe `METERING_ROUTE_HOST` or `METERING_ROUTE_HOSTNAME`?

Comment (timflannagan1) [reply Documentation/configuring-reporting-operator.md]: Yeah, that was a poor choice for that environment variable name - will update to `METERING_ROUTE_HOSTNAME`.

File: Documentation/writing-custom-queries.md [hunk 2]
```diff
@@ -343,15 +343,13 @@ kubectl -n $METERING_NAMESPACE get report unready-deployment-replicas -o json
...
-```
-If you are using Openshift, you'll need to change the url to the following:
-```
-http://127.0.0.1:8001/api/v1/namespaces/$METERING_NAMESPACE/services/https:reporting-operator:http/proxy/api/v1/reports/get?name=unready-deployment-replicas&namespace=$METERING_NAMESPACE&format=csv
+METERING_ROUTE_HOSTNAME=$(oc -n $METERING_NAMESPACE get routes metering -o json | jq '.status.ingress[].host')
+TOKEN=$(oc -n $METERING_NAMESPACE serviceaccounts get-token reporting-operator)
+curl -H "Authorization: Bearer $TOKEN" -k "https://$METERING_ROUTE_HOSTNAME/api/v1/reports/get?name=unready-deployment-replicas&namespace=$METERING_NAMESPACE&format=csv"
 ```
 
+If you aren't using Openshift, or have disabled `spec.tls.enabled` top-level key, follow the [viewing reports](viewing-reports) section, and come back to this document afterwards.
```
Comment (chancez) [inline Documentation/writing-custom-queries.md]: This should be `[viewing reports][viewing-reports]`

Comment (timflannagan1) [reply Documentation/writing-custom-queries.md]: Good catch, updated.

**Pull request merged**